### ISSUE TITLE
Aerial Sporting and Recreational Activity

### ISSUE THREAD

**Issue by Pops-67** [openAIP/openaip#169]
Aerial Sporting and Recreational Activity is the standard name for this kind of activity and the name of the corresponding airspace. This is then specified more as to what kind of activity/hazard this airspace contain. This could be skydiving, paragliders/hang-gliders, balloon and so on - and gliding. OpenAIP has only "Gliding", and names the airspace as such. This is not correct unless the activity is gliding.

**Comment by reskume** [comment]
The current way of creating "arial sporting and recreational activities" is correct. Those airspaces are defined in (for Germany) ED ENR 5.5. Depending on the type of airspace, you can either choose ICAO class "SUA - Special User Airspace" and a specific type. Most likely, the type would either be "Restricted", "Warning", "Danger" or "Prohibited" with a meaningful name of the airspace. The "Remarks" and "Hours of operations" fields can also be used to add more metadata and be very specific about the intended use case for the airspace. Another special airspace construct (in Germany) are "gliding sectors" aka "Segelflugsektoren". Those are regionally defined and sometimes even only apply to specific airports regulated by a special agreement. Those airspaces are used to activate a specific portion of an encasing airspace on demand to "switch" its class so that gliding activities are legally possible. See [DE - Segelflugsektoren](https://www.daec.de/fachbereiche/luftraum-flugbetrieb/segelflugsektoren/) for reference. 

TLDR: When creating e.g. "Parachuting Zones" or "Aerobatic Boxes" use ICAO class "SUA" with specific type (Warning, Restricted). For gliding sectors, use ICAO class "SUA" and "gliding sector".

**Comment by Pops-67** [comment]
The official AIPs see this differently. The standard structure is:ENR 5.1 Prohibited, restricted and danger areas
ENR 5.2 Military exercise and training areas and air defence identification zone (ADIZ)

ENR 5.3.1 Other activities of a dangerous nature
ENR 5.3.2 Other regulated airspaces
ENR 5.5 Aerial sporting and recreational activities

**Comment by reskume** [comment]
Hm, I don't see where this differs from the way openaip defines airspaces. The "Aerial sporting and recreational activities" are a set of airspaces that can be already created using for example see [Parachuting Zone Herrenteich](https://www.openaip.net/data/airspaces/62614af4cb27f42509448772). Gliding sectors behave differently thus they have their own special use airspace "type".

AFAIK the gliding sectors are NOT defined in the AIP. Correct me if I'm wrong.

**Comment by reskume** [comment]
Evaluate adding new special airspace type "AERIAL_SPORTING_RECREATIONAL". This would require updating the model to also have an additional property "intendedUsage" that describes the intended usage of an airspace.

- [ ] update model to have new type
- [ ] update model to have new "intendedUsage" property - default "none"
- [ ] update validators to require "intendedUsage" for all airspaces
- [ ] update schemas for CRUD operations to reflect changes in airspace model

Possible show stoppers for this special airspace are:

- all airspaces of this type are defined as either "R" or "Q" in import sources so it's only possible to create them via the web UI
- bulk imports require manually updating this special type airspaces which will add a lot of maintenance overhead

**Comment by Pops-67** [comment]
OK, maybe you are right. It's just a bit confusing (to me) when things are structurally different, yet the same data shall be displayed.

**Comment by Pops-67** [comment]
Sorry for all the comments. Looked at it again, and I see the main problem. SUA is OK. However among the choices of SUAs, the only one that could fit (somewhat) is Alert Aera. Protected, danger, prohibited, restricted are all well defined, and are definitely not areas for air sporting and recreational activities. Which is the reason why PANS-AIM defines a separate air sports and recreational activities area, with specific info associated with that air space.

**Comment by reskume** [comment]
The most straight forward way to implement this would be to add new "types" to the airspace, like "Parachuting (Sporting/Recreational), "Arial Work (Sporting/Recreational)", "UL Corridors (Sporting/Recreational)" etc. Would do you think, would this be sufficient?

**Comment by Pops-67** [comment]
That would work I guess. However, it would not be in accordance with what PANS-AIM defines as info fields for this airspace. It may therefore not be what one can find in the official AIPs (additional fields for information would be required). AIXM 5 has an "activity" field which can be used to describe the actual activity (parachuting, aerial work etc). The AIPs normally use the "Operator" field with contact information (which implicitly describes the activity). But, I mean, what is useful for a GA pilot? To know the activity definitely is useful. The "operator" may be useful (even crucial), or it may be completely irrelevant depending on your specific GA activity. On the other hand, following the official structure of PANS-AIM and AIXM 5, which is what the official AIPs use, none of this would be an issue whatsoever ;-)

**Comment by Pops-67** [comment]
I see now that the use of SUA is not correct. There are only 7 classes of airspace: A, B, C, D, E, F and G (it can also be unclassified, but that's beside the point and irrelevant in this context). All of these airspaces can have SUA within, but this doesn't change the _class_ of airspace to SUA, because SUA is not a class. The class is still whatever originally was there. SUA is more of an exception to a pocket of airspace. The error done here is that this pocket loses its class. Instead of being C, D etc it becomes "SUA", and this is not correct.

**Comment by reskume** [comment]
> I see now that the use of SUA is not correct. There are only 7 classes of airspace: A, B, C, D, E, F and G (it can also be unclassified, but that's beside the point and irrelevant in this context). All of these airspaces can have SUA within, but this doesn't change the _class_ of airspace to SUA, because SUA is not a class. The class is still whatever originally was there. SUA is more of an exception to a pocket of airspace. The error done here is that this pocket loses its class. Instead of being C, D etc it becomes "SUA", and this is not correct.

I can understand your point. This may get a little off-topic now and we should move such discussions to https://github.com/openAIP/openaip/discussions or open another issue. Anyway, the SUA class is a required "utility class" since a lot (if not all) of import sources are openAIR formatted. In an ideal world, i.e. all input is done via OpenAIP, the SUA class would not be required but since openAIR sources define e.g. a glider sector with a simple "AC W", it's impossible to actually map its correct ICAO class. For those cases, all "AC" openAIR definitions that are NOT simple ICAO classes have an internal mapping for most commonly used type -> classes if possible (e.g RMZ / TMZ / ATZ etc => this is also not ideal and may be not correct for every country) but many other can only be mapped to the SUA class.

**Comment by reskume** [comment]
I would propose the following changes to the airspace model:

- [ ] Update the model to have a new property "intendedActivity" to describe the intended usage (activity) for an airspace. The possible values could be "none" as default for all "non-activity-based" airspaces + extendable list of activities.
- [ ] an additional specific type "AERIAL_SPORTING_RECREATIONAL"

The ICAO class can either be mapped to SUA or any other ICAO class that is suitable.
